domain.go

```diff
@@ -0,0 +1,145 @@
+package libvirt
+
+import (
+	"encoding/xml"
+
+	libvirtxml "github.com/libvirt/libvirt-go-xml"
+)
+
+func domainXML(d *Driver) (string, error) {
+	domain := libvirtxml.Domain{
+		Type: "kvm",
+		Name: d.MachineName,
+		Memory: &libvirtxml.DomainMemory{
+			Value: uint(d.Memory),
+			Unit:  "MB",
+		},
+		VCPU: &libvirtxml.DomainVCPU{
+			Placement: "static",
+			Value:     uint(d.CPU),
+		},
+		Features: &libvirtxml.DomainFeatureList{
+			ACPI: &libvirtxml.DomainFeature{},
+			APIC: &libvirtxml.DomainFeatureAPIC{},
+			PAE:  &libvirtxml.DomainFeature{},
+		},
+		CPU: &libvirtxml.DomainCPU{
+			Mode: "host-passthrough",
+			Features: []libvirtxml.DomainCPUFeature{
+				{
+					Policy: "disable",
+					Name:   "rdrand",
+				},
+			},
+		},
+		OS: &libvirtxml.DomainOS{
+			Type: &libvirtxml.DomainOSType{
+				Arch: "x86_64",
+				Type: "hvm",
+			},
+			BootDevices: []libvirtxml.DomainBootDevice{
+				{
+					Dev: "hd",
+				},
+			},
+			BootMenu: &libvirtxml.DomainBootMenu{
+				Enable: "no",
+			},
+		},
+		Clock: &libvirtxml.DomainClock{
+			Offset: "utc",
+		},
+		OnPoweroff: "destroy",
+		OnReboot:   "restart",
+		OnCrash:    "destroy",
```

Comment:
Fwiw, I think some of these fields can be omitted as these are the default  values libvirt will use, but this can be addressed later.

Reply:
yes I confirm